src/components/structures/MatrixChat.js

```diff
@@ -693,6 +696,16 @@ module.exports = React.createClass({
     },
 
     /**
+     * Called when the team token is acquired at registration or login. This
+     * should only happen if the configured RTS (see config.json teamServerURL)
+     * recognises the user as having an email address that matches a team.
+     */
+    _onSetTeamToken: function(teamToken) {
+        this._teamToken = teamToken;
+        this._setPage(PageTypes.HomePage);
```

Comment:
Is it necessary to set the page here? It's not intuitive to me that it should jump to your welcome page when a  team token is set